corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -486,19 +494,34 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
         getIndexedRowDataList: function () {
             let indexedRowDataList = [];
             for (let model of this.options.groupModelsList) {
-                let indexedRowData = model.get('data')
-                    .reduce((acc, data, i) => {
-                        if (this.options.bodyRowIndices.includes(i)) {
-                            acc[i] = data;
-                        }
-                        return acc;
-                    }, {});
-                if (Object.keys(indexedRowData).length !== 0) {
-                    indexedRowDataList.push(indexedRowData);
+                if (model.id === this.model.get('updatedCaseId')) {
+                    indexedRowDataList.push(this.model.get('updatedRowData'));
+                } else {
+                    let indexedRowData = model.get('data')
+                        .reduce((acc, data, i) => {
+                            if (this.options.bodyRowIndices.includes(i)) {
+                                acc[i] = data;
+                            }
+                            return acc;
+                        }, {});
+                    if (Object.keys(indexedRowData).length !== 0) {
+                        indexedRowDataList.push(indexedRowData);
+                    }
                 }
             }
             return indexedRowDataList;
         },
+
+        updateModelFromDetailResponse: function (caseId, detailResponse) {
+            this.model.set('updating', true)
+            CaseTileGroupedView.__super__.updateModelFromDetailResponse.apply(this, [caseId, detailResponse])
+            this.model.set('updatedCaseId', caseId);
+            this.model.set('updatedRowData', this.options.bodyRowIndices.reduce((acc, index) => {
```

Comment:
The problem was that `groupModelsList` is in the options and not the model. I don't think you can update those after the fact. But please correct me if I am wrong on that. I agree it would be cleaner to update `groupModelsList`.

Reply:
hm good point, I'm not entirely sure about the lifecycle of `render()`. You might be right that `render` would override changes to `groupModelsList` with new values from `childViewOptions`